nixos/modules/virtualisation/docker.nix

```diff
@@ -196,7 +197,7 @@ in
         };
 
         path = [ pkgs.kmod ] ++ optional (cfg.storageDriver == "zfs") pkgs.zfs
-          ++ optional cfg.enableNvidia pkgs.nvidia-docker
+          ++ optional (cfg.enableNvidia && !cfg.package.supports-cdi) pkgs.nvidia-docker
```

Comment:
My personal preference: `optionals [x]` over `optional x`

Reply:
I would like to not introduce unrelated changes.